Sources/StreamChatUI/ChatMessageList/Reactions/ChatMessageReactionsView.swift

```diff
@@ -7,6 +7,122 @@ import UIKit
 
 public typealias ChatMessageReactionsView = _ChatMessageReactionsView<NoExtraData>
 
+class ReactionsBubbleView<ExtraData: ExtraDataTypes>: _View, UIConfigProvider {
+    private let tailHeight: CGFloat = 6
+
+    var tailDirection: _ChatMessageThreadArrowView<ExtraData>.Direction? {
+        didSet {
+            updateContentIfNeeded()
+        }
+    }
+
+    override func defaultAppearance() {
+        super.defaultAppearance()
+
+        backgroundColor = .clear
+    }
+
+    override func draw(_ rect: CGRect) {
+        super.draw(rect)
+
+        strokeColor?.setStroke()
+        fillColor?.setFill()
+
+        let bubbleAndTail = bubblePath()
+        bubbleAndTail.stroke()
+        bubbleAndTail.fill()
+    }
+
+    override func setUpLayout() {
+        directionalLayoutMargins.bottom += tailHeight
+    }
+
+    override func updateContent() {
+        setNeedsDisplay()
+    }
+
+    var maskingPath: UIBezierPath {
+        bubblePath(withRadiiIncreasedBy: 4)
+    }
+}
+
+private extension ReactionsBubbleView {
+    var fillColor: UIColor? {
+        tailDirection.map {
+            $0 == .toTrailing ?
+                uiConfig.colorPalette.popoverBackground :
+                uiConfig.colorPalette.background2
+        }
+    }
+
+    var strokeColor: UIColor? {
+        tailDirection.map {
+            $0 == .toTrailing ?
+                uiConfig.colorPalette.border :
+                uiConfig.colorPalette.background2
+        }
+    }
+
+    var bubbleBodyCenter: CGPoint {
+        bounds
+            .inset(by: .init(top: 0, left: 0, bottom: tailHeight, right: 0))
+            .center
+    }
+
+    var bigTailCirleCenter: CGPoint {
+        bubbleBodyCenter.offsetBy(
+            dx: tailDirection == .toTrailing ? 10 : -10,
+            dy: 14
+        )
+    }
+
+    var smallTailCirleCenter: CGPoint {
+        bigTailCirleCenter.offsetBy(
+            dx: tailDirection == .toTrailing ? 4 : -4,
+            dy: 6
+        )
+    }
```

Comment:
In `ReactionsBubbleView` there is a plenty of magic numbers (circle radii and offsets). 

Was trying to get rid of them but didn't succeed since **we have to have shape centres fixed so we can make scale them for the mask and they stay in the same place**